primitives/src/merkle_tree/field_based_mht/optimized/mod.rs

```diff
@@ -40,16 +40,25 @@ impl<T: BatchFieldBasedMerkleTreeParameters> FieldBasedOptimizedMHT<T> {
     /// the batch hash: you can edit the `mht_poseidon_tuning` benchmarks in
     /// `primitives/src/benches/poseidon_mht.rs` to properly tune the `processing_step`
     /// parameter according to your use case.
-    pub fn init(height: usize, processing_step: usize) -> Self {
-        assert!(check_precomputed_parameters::<T>(height));
+    pub fn init(height: usize, processing_step: usize) -> Result<Self, Error> {
+        // TODO: check error message
+        if !check_precomputed_parameters::<T>(height) {
+            return Err(Box::new(MerkleTreeError::Other("precomputed parameters check failed".to_owned())));
```

Comment:
("Unsupported height. Max supported height is: {}", T::ZERO_NODE_CST.unwrap().nodes.len())